applications/nrf_desktop/src/modules/qos.c

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <zephyr.h>
+#include <zephyr/types.h>
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/gatt.h>
+
+#include "chmap_filter.h"
+
+#include "ble_event.h"
+
+#define MODULE qos
+#include "module_state_event.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(MODULE, CONFIG_DESKTOP_BAS_LOG_LEVEL);
+
+
+static struct bt_uuid_128 custom_desc_uuid = BT_UUID_INIT_128(
+		0x4c, 0x65, 0xe6, 0xe6, 0x98, 0xb6, 0x78, 0xd5,
+		0x7d, 0x17, 0x19, 0x46, 0x75, 0x2, 0x47, 0x24);
+static const struct bt_uuid_128 custom_desc_chrc_uuid = BT_UUID_INIT_128(
+		0x4c, 0x65, 0xe6, 0xe6, 0x98, 0xb6, 0x78, 0xd5,
+		0x7d, 0x17, 0x19, 0x46, 0x75, 0x2, 0x47, 0x24);
+
+static bool active;
+static u8_t chmap[CHMAP_BLE_BITMASK_SIZE];
+static struct k_spinlock lock;
+
+static void chmap_ccc_cfg_changed(const struct bt_gatt_attr *attr,
+				 u16_t value)
+{
```

Comment:
We should use spinlock for active too

Reply:
Or make active as atomic

Reply:
we can to be formally correct but it does not matter on our arch
bas has the same problem btw